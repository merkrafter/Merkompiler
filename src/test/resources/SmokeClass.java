// no public classes allowed
class SmokeClass {
    // no 'static' keyword available
    // no 'String' type available
    public void main() {
        int a;
        int b;
        int c;
        int d;
        int e;
        int result;
        a = 1;
        b = 2;
        c = 3;
        d = 4;
        e = 5;
        result = a+(b-c)*d/e;
        if (a <= b) {println(a);} else {println(b);}
        if (a <  b) {println(a);} else {println(b);}
        if (a == b) {println(a);} else {println(b);}
        if (a  > b) {println(a);} else {println(b);}
        if (a >= b) {println(a);} else {println(b);}
        println(result);
    }
}
