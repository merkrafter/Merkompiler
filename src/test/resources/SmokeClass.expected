SmokeClass.java(2,1): KEYWORD(class)
SmokeClass.java(2,7): IDENT(SmokeClass)
SmokeClass.java(2,18): L_BRACE
SmokeClass.java(5,5): KEYWORD(public)
SmokeClass.java(5,12): KEYWORD(void)
SmokeClass.java(5,17): IDENT(main)
SmokeClass.java(5,21): L_PAREN
SmokeClass.java(5,22): R_PAREN
SmokeClass.java(5,24): L_BRACE
SmokeClass.java(6,9): KEYWORD(int)
SmokeClass.java(6,13): IDENT(a)
SmokeClass.java(6,14): SEMICOLON
SmokeClass.java(7,9): KEYWORD(int)
SmokeClass.java(7,13): IDENT(b)
SmokeClass.java(7,14): SEMICOLON
SmokeClass.java(8,9): KEYWORD(int)
SmokeClass.java(8,13): IDENT(c)
SmokeClass.java(8,14): SEMICOLON
SmokeClass.java(9,9): KEYWORD(int)
SmokeClass.java(9,13): IDENT(d)
SmokeClass.java(9,14): SEMICOLON
SmokeClass.java(10,9): KEYWORD(int)
SmokeClass.java(10,13): IDENT(e)
SmokeClass.java(10,14): SEMICOLON
SmokeClass.java(11,9): KEYWORD(int)
SmokeClass.java(11,13): IDENT(result)
SmokeClass.java(11,19): SEMICOLON
SmokeClass.java(12,9): IDENT(a)
SmokeClass.java(12,11): ASSIGN
SmokeClass.java(12,13): NUMBER(1)
SmokeClass.java(12,14): SEMICOLON
SmokeClass.java(13,9): IDENT(b)
SmokeClass.java(13,11): ASSIGN
SmokeClass.java(13,13): NUMBER(2)
SmokeClass.java(13,14): SEMICOLON
SmokeClass.java(14,9): IDENT(c)
SmokeClass.java(14,11): ASSIGN
SmokeClass.java(14,13): NUMBER(3)
SmokeClass.java(14,14): SEMICOLON
SmokeClass.java(15,9): IDENT(d)
SmokeClass.java(15,11): ASSIGN
SmokeClass.java(15,13): NUMBER(4)
SmokeClass.java(15,14): SEMICOLON
SmokeClass.java(16,9): IDENT(e)
SmokeClass.java(16,11): ASSIGN
SmokeClass.java(16,13): NUMBER(5)
SmokeClass.java(16,14): SEMICOLON
SmokeClass.java(17,9): IDENT(result)
SmokeClass.java(17,16): ASSIGN
SmokeClass.java(17,18): IDENT(a)
SmokeClass.java(17,19): PLUS
SmokeClass.java(17,20): L_PAREN
SmokeClass.java(17,21): IDENT(b)
SmokeClass.java(17,22): MINUS
SmokeClass.java(17,23): IDENT(c)
SmokeClass.java(17,24): R_PAREN
SmokeClass.java(17,25): TIMES
SmokeClass.java(17,26): IDENT(d)
SmokeClass.java(17,27): DIVIDE
SmokeClass.java(17,28): IDENT(e)
SmokeClass.java(17,29): SEMICOLON
SmokeClass.java(18,9): KEYWORD(if)
SmokeClass.java(18,12): L_PAREN
SmokeClass.java(18,13): IDENT(a)
SmokeClass.java(18,16): LOWER_EQUAL
SmokeClass.java(18,18): IDENT(b)
SmokeClass.java(18,19): R_PAREN
SmokeClass.java(18,21): L_BRACE
SmokeClass.java(18,22): IDENT(println)
SmokeClass.java(18,29): L_PAREN
SmokeClass.java(18,30): IDENT(a)
SmokeClass.java(18,31): R_PAREN
SmokeClass.java(18,32): SEMICOLON
SmokeClass.java(18,33): R_BRACE
SmokeClass.java(18,35): KEYWORD(else)
SmokeClass.java(18,40): L_BRACE
SmokeClass.java(18,41): IDENT(println)
SmokeClass.java(18,48): L_PAREN
SmokeClass.java(18,49): IDENT(b)
SmokeClass.java(18,50): R_PAREN
SmokeClass.java(18,51): SEMICOLON
SmokeClass.java(18,52): R_BRACE
SmokeClass.java(19,9): KEYWORD(if)
SmokeClass.java(19,12): L_PAREN
SmokeClass.java(19,13): IDENT(a)
SmokeClass.java(19,15): LOWER
SmokeClass.java(19,18): IDENT(b)
SmokeClass.java(19,19): R_PAREN
SmokeClass.java(19,21): L_BRACE
SmokeClass.java(19,22): IDENT(println)
SmokeClass.java(19,29): L_PAREN
SmokeClass.java(19,30): IDENT(a)
SmokeClass.java(19,31): R_PAREN
SmokeClass.java(19,32): SEMICOLON
SmokeClass.java(19,33): R_BRACE
SmokeClass.java(19,35): KEYWORD(else)
SmokeClass.java(19,40): L_BRACE
SmokeClass.java(19,41): IDENT(println)
SmokeClass.java(19,48): L_PAREN
SmokeClass.java(19,49): IDENT(b)
SmokeClass.java(19,50): R_PAREN
SmokeClass.java(19,51): SEMICOLON
SmokeClass.java(19,52): R_BRACE
SmokeClass.java(20,9): KEYWORD(if)
SmokeClass.java(20,12): L_PAREN
SmokeClass.java(20,13): IDENT(a)
SmokeClass.java(20,16): EQUAL
SmokeClass.java(20,18): IDENT(b)
SmokeClass.java(20,19): R_PAREN
SmokeClass.java(20,21): L_BRACE
SmokeClass.java(20,22): IDENT(println)
SmokeClass.java(20,29): L_PAREN
SmokeClass.java(20,30): IDENT(a)
SmokeClass.java(20,31): R_PAREN
SmokeClass.java(20,32): SEMICOLON
SmokeClass.java(20,33): R_BRACE
SmokeClass.java(20,35): KEYWORD(else)
SmokeClass.java(20,40): L_BRACE
SmokeClass.java(20,41): IDENT(println)
SmokeClass.java(20,48): L_PAREN
SmokeClass.java(20,49): IDENT(b)
SmokeClass.java(20,50): R_PAREN
SmokeClass.java(20,51): SEMICOLON
SmokeClass.java(20,52): R_BRACE
SmokeClass.java(21,9): KEYWORD(if)
SmokeClass.java(21,12): L_PAREN
SmokeClass.java(21,13): IDENT(a)
SmokeClass.java(21,16): GREATER
SmokeClass.java(21,18): IDENT(b)
SmokeClass.java(21,19): R_PAREN
SmokeClass.java(21,21): L_BRACE
SmokeClass.java(21,22): IDENT(println)
SmokeClass.java(21,29): L_PAREN
SmokeClass.java(21,30): IDENT(a)
SmokeClass.java(21,31): R_PAREN
SmokeClass.java(21,32): SEMICOLON
SmokeClass.java(21,33): R_BRACE
SmokeClass.java(21,35): KEYWORD(else)
SmokeClass.java(21,40): L_BRACE
SmokeClass.java(21,41): IDENT(println)
SmokeClass.java(21,48): L_PAREN
SmokeClass.java(21,49): IDENT(b)
SmokeClass.java(21,50): R_PAREN
SmokeClass.java(21,51): SEMICOLON
SmokeClass.java(21,52): R_BRACE
SmokeClass.java(22,9): KEYWORD(if)
SmokeClass.java(22,12): L_PAREN
SmokeClass.java(22,13): IDENT(a)
SmokeClass.java(22,16): GREATER_EQUAL
SmokeClass.java(22,18): IDENT(b)
SmokeClass.java(22,19): R_PAREN
SmokeClass.java(22,21): L_BRACE
SmokeClass.java(22,22): IDENT(println)
SmokeClass.java(22,29): L_PAREN
SmokeClass.java(22,30): IDENT(a)
SmokeClass.java(22,31): R_PAREN
SmokeClass.java(22,32): SEMICOLON
SmokeClass.java(22,33): R_BRACE
SmokeClass.java(22,35): KEYWORD(else)
SmokeClass.java(22,40): L_BRACE
SmokeClass.java(22,41): IDENT(println)
SmokeClass.java(22,48): L_PAREN
SmokeClass.java(22,49): IDENT(b)
SmokeClass.java(22,50): R_PAREN
SmokeClass.java(22,51): SEMICOLON
SmokeClass.java(22,52): R_BRACE
SmokeClass.java(23,9): IDENT(println)
SmokeClass.java(23,16): L_PAREN
SmokeClass.java(23,17): IDENT(result)
SmokeClass.java(23,23): R_PAREN
SmokeClass.java(23,24): SEMICOLON
SmokeClass.java(24,5): R_BRACE
SmokeClass.java(25,1): R_BRACE
SmokeClass.java(26,0): EOF
